code/game/gamemodes/blob/blobs/shield.dm

```diff
@@ -38,26 +38,16 @@
 	brute_resist = 0.5
 	explosion_block = 2
 	point_return = 9
-	flags_2 = CHECK_RICOCHET_1
-	var/reflect_chance = 80 //80% chance to reflect
+	flags_2 = CHECK_RICOCHET_2
 
 /obj/structure/blob/shield/reflective/handle_ricochet(obj/item/projectile/P)
-	if(P.is_reflectable && prob(reflect_chance) && !P.legacy)
-		var/P_turf = get_turf(P)
-		var/face_direction = get_dir(src, P_turf)
-		var/face_angle = dir2angle(face_direction)
-		var/incidence_s = GET_ANGLE_OF_INCIDENCE(face_angle, (P.Angle + 180))
-		if(abs(incidence_s) > 90 && abs(incidence_s) < 270)
-			return FALSE
-		var/new_angle_s = SIMPLIFY_DEGREES(face_angle + incidence_s)
-		P.setAngle(new_angle_s)
-		P.firer = src //so people who fired the lasers are not immune to them when it reflects

```

Comment:
Really. Let's set the firer, which is always a mob to be an object, what could go wrong? Never runtimes like `undefined var: /obj/structure/blob/shield/reflective/var/ckey` No, that would never happen, oh no no.